src/stash-list/bookmark.vue

```diff
@@ -20,7 +20,16 @@
      @auxclick.middle.exact.prevent.stop="closeOrHideOrOpen">
      {{bookmark.title}}
   </a>
+  <input
+    ref="newtitle"
+    v-show="isEditable"
+    v-model="bookmark.title"
```

Comment:
If you use `v-model` here, Vue will edit the title on its own, which is probably not what you want (even though it's what all the Vue documentation tells you to do...), and then there is no easy way to cancel the rename and revert to the original value.

It turns out this is unfortunately quite a hard problem to get right; I ended up writing a whole component to deal with it for renaming folders/stashes: https://github.com/josh-berry/tab-stash/blob/master/src/components/editable-label.vue

We might be able to refactor editable-label a bit so you can use it here as well. I think the main challenge is going to be how to toggle editing from outside the component. If refactoring turns out to be untenable/difficult, then maybe it will at least give you some ideas on how to proceed here.

Reply:
haha, by coincidence, about while you were writing this I was thinking about how to handle canceled edits, and pushed an update for that just before seeing this. My solution is probably less elegant than adapting the editable-label component (I hadn't spotted that one), but is simple to implement and I haven't spotted any missed edge-cases yet - what do you think?

Reply:
A couple edge cases I could think of:

* If the bookmark is renamed by some other method while the text box is open (e.g. Firefox Sync, some other part of the UI, etc.), the bookmark model might get stuck with the old value and never see the updated name, even if the user cancels editing.
* Some of the smaller behavioral details (e.g. select-all on focus, the OK/Cancel buttons) don't have consistent behavior with editable-label, which might get a bit confusing. And just from a clean-code/maintenance perspective, it's probably better to have all this behavior in one place.

I took a stab this morning at rewriting editable-label to something you could hopefully drop in and have it Just Work for bookmarks as well:

https://github.com/josh-berry/tab-stash/pull/263

If you look at `src/stash-list/folder.vue`, you'll see how it can be used. What do you think?

Reply:
looks like it works pretty well, I've merged that branch into mine and refactored this to use async-text-input